src/Table/Snowflake/SnowflakeTableReflection.php

```diff
@@ -0,0 +1,197 @@
+            sprintf(
+                'DESC TABLE %s',
+                SnowflakeQuote::createQuotedIdentifierFromParts([$this->schemaName, $this->tableName,])
+            )
+        );
+
+        $columns = [];
+
+        foreach ($columnsMeta as $col) {
+            if ($col['kind'] === 'COLUMN') {
+                $columns[] = SnowflakeColumn::createFromDB($col);
+            }
+        }
+
+        return new ColumnCollection($columns);
+    }
+
+
+
+    public function getRowsCount(): int
+    {
+        $result = $this->connection->fetchOne(sprintf(
+            'SELECT COUNT(*) AS NumberOfRows FROM %s',
+            SnowflakeQuote::createQuotedIdentifierFromParts([
+                $this->schemaName,
+                $this->tableName,
+            ])
+        ));
+        return (int) $result;
+    }
+
+    /**
+     * returns list of column names where PK is defined on
+     *
+     * @return string[]
+     */
+    public function getPrimaryKeysNames(): array
+    {
+        $columnsMeta = $this->connection->fetchAllAssociative(
+            sprintf(
+                'SHOW PRIMARY KEYS IN TABLE %s',
+                SnowflakeQuote::createQuotedIdentifierFromParts([$this->schemaName, $this->tableName,])
+            )
+        );
+
+        return array_map(function ($pkRow) {
+            return $pkRow['column_name'];
+        }, $columnsMeta);
+    }
+
+    public function getTableStats(): TableStatsInterface
+    {
+        $sql = sprintf(
+            'SHOW TABLES LIKE %s IN SCHEMA %s',
+            ExasolQuote::quote($this->tableName),
+            ExasolQuote::quoteSingleIdentifier($this->schemaName)
+        );
+        $result = $this->connection->fetchAssociative($sql);
+        $bytes = $result ? (int) $result['bytes'] : 0;
+        return new TableStats($bytes, $this->getRowsCount());
+    }
+
+    public function isTemporary(): bool
+    {
+        return $this->isTemporary;
+    }
+
+    /**
+     * @return array<int, array<string, mixed>>
+     * array{
+     *  schema_name: string,
+     *  name: string
+     * }[]
+     */
+    public function getDependentViews(): array
+    {
+        $databaseName = $this->connection->fetchOne('SELECT CURRENT_DATABASE()');
+        $views = $this->connection->fetchAllAssociative(
+            sprintf(
+                'SHOW VIEWS IN DATABASE %s',
+                SnowflakeQuote::quoteSingleIdentifier($databaseName)
+            )
+        );
+
+        $dependentViews = [];
+        foreach ($views as $viewRow) {
+            // check that the tableName exists in DDL of the view
+            if (preg_match('/.*' . $this->tableName . '.*/i', $viewRow['text']) === 1) {
+                try {
+                    $dependentObjects = $this->connection->fetchAllAssociative(
+                        sprintf(
+                            '
+SELECT * FROM TABLE(get_object_references(database_name=>%s, SCHEMA_NAME=>%s, object_name=>%s))  
```

Comment:
je to masakr, ale na nic lepsiho jsem neprisel

Reply:
A vlastně tím trochu rozbíjím všecko kde je SHOW query, protože možeš mět ve SNFLK dva objekty třeba tabulky `TABLE1` a `"table1"` když uděláš SHOW tak ten je case insensitive a vylistuje oba případy. Takže to budeš muset ošetřit a napsat na to testy.
A možeš si za to sám, protože kdybys nevymýšlal takové kraviny tak bych na to nedošel :D